partner_export_wamas/models/res_partner.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def _wamas_export(self, specific_dict=None, telegram=False):
+        """
+        Export the partner data as WAMAS format
+
+        :return: partner data as WAMAS format
+        :rtype: bytes
+        """
+        self.ensure_one()
+        if not telegram:
+            raise ValueError(_("Need telegram for exporting"))
+        # If having a specific dict for the partner, we use it
+        if specific_dict and isinstance(specific_dict, dict):
+            dict_partner = specific_dict
+        else:
+            dict_partner = self.read()[0]
```

Comment:
`self.read()` will read too many things. It can be quite huge.
IMO we should use cached data and reduce the fields to read somehow. 
IF you don't know yet what to do and you are only using the specific_dict just remove it and make the specific_dict mandatory.